### PR TITLE
feat(ivy): support injecting the injector

### PR DESCRIPTION
**Pull request by kara**
This PR adds support for injecting the Injector itself in both the runtime and the compiler. The compiler changes should also fix a number of TestBed tests failing under Ivy with `Error: No value provided for @angular/core symbol 'INJECTOR'`.

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by kara** [comment]
[presubmit](https://test.corp.google.com/ui#id=OCL:218398417:BASE:218398490:1540327150801:ef219c)

File: packages/core/test/render3/di_spec.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {Attribute, ChangeDetectorRef, ElementRef, Host, InjectFlags, Injector, O
 import {RenderFlags} from '@angular/core/src/render3/interfaces/definition';
 
 import {defineComponent} from '../../src/render3/definition';
-import {bloomAdd, bloomHashBitOrFactory as bloomHash, getOrCreateNodeInjector, injectAttribute, injectorHasToken} from '../../src/render3/di';
+import {bloomAdd, bloomHashBitOrFactory as bloomHash, getOrCreateNodeInjector, injectAttribute, injectorHasToken,} from '../../src/render3/di';
```
Comment (alfaproject) [inline packages/core/test/render3/di_spec.ts]: Revert?

**Review by mhevery — APPROVED** [review]

**Comment by matsko** [comment]
@kara please rebase

**Comment by matsko** [comment]
@kara please rebase

**Pull request closed without merging**